src/oc_erchef/apps/chef_license/src/chef_license_worker.erl

```diff
@@ -0,0 +1,128 @@
+-module(chef_license_worker).
+
+-behaviour(gen_server).
+
+-include("../../../include/chef_types.hrl").
+
+-export([
+    start_link/0,
+    get_license/0,
+    test_license/0
+]).
+
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    code_change/2,
+    terminate/2
+]).
+
+-record(state, {
+    scanned_time,
+    grace_period,
+    license_cache,
+    message
+}).
+
+-define(DEFAULT_LICENSE_SCAN_INTERVAL, 3000). %milli seconds
+
+-define(DEFAULT_FILE_PATH, "/tmp/lic").
+
+-define(LICENSE_SCAN_CMD, "chef-automate license status --result-json ").
+
+
+%%% ======================================
+%%% Exported
+%%% ======================================
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+get_license()->
+    gen_server:handle_call(?MODULE, get_license).
+
+%%% ======================================
+%%% Gen Server callbacks
+%%% ======================================
+init(_Config) ->
+    State = check_license(#state{}),
+    gen_server:send_after(?DEFAULT_LICENSE_SCAN_INTERVAL, check_license),
+    {ok, State}.
+
+handle_call(get_license, _From, #state{license_cache = Lic, grace_period = GracePeriod, message = Msg} = State) ->
+    {reply,{Lic, GracePeriod, Msg}, State};
+
+handle_call(_Message, _From, State) ->
+    {noreply, State}.
+
+handle_cast(_Message, State) ->
+    {noreply, State}.
+
+handle_info(check_license, State)->
+    check_license(State),
+    gen_server:send_after(?DEFAULT_LICENSE_SCAN_INTERVAL, check_license),
+    {noreply,State};
+
+handle_info(_Message, State) ->
+    {noreply, State}.
+
+code_change(_OldVsn, State) ->
+    {noreply, State}.
+
+terminate(_Reason, _State) ->
+    ok.
+
+%%% =====================
+%%% Internal functions
+%%% =====================
+check_license(State) ->
+    JsonStr = get_license_info(),
+    case process_license(JsonStr) of
+        {ok, valid_license} -> State#state{license_cache=valid, scanned_time = erlang:timestamp()};
+        {ok, expired} -> State#state{license_cache=expired, scanned_time = erlang:timestamp()};
+        {ok, valid_license, grace_period} -> State#state{license_cache=valid, grace_period=true, scanned_time = erlang:timestamp()};
+        {error, invalid_response} -> State
+    end.
+
+get_license_info() ->
+    os:cmd(?LICENSE_SCAN_CMD ++ ?DEFAULT_FILE_PATH),
```

Comment:
Instead of writing to a file, can you parse the output directly from os:cmd?

Reply:
a: no, json output requires a file path.

Reply:
`chef-automate license status --result-json` needs file argument so the direct output is not having all the details.